src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -1416,6 +1416,37 @@ private static bool CanResetFitViewToggle(object o)
             return true;
         }
 
+        /// <summary>
+        /// Selects an Element by ID and focuses the view around it
+        /// </summary>
+        /// <param name="id"></param>
+        private void FocusNode(object id)
+        {
+            try
+            {
+                var node = DynamoViewModel.Model.CurrentWorkspace.Nodes.First(x => x.GUID.ToString() == id.ToString());
+                if(node != null)
+                {
+                    //select the element
+                    DynamoSelection.Instance.ClearSelection();
+                    DynamoSelection.Instance.Selection.Add(node);
+
+                    DynamoViewModel.ShowElement(node, false);
```

Comment:
Why call `ShowElement` at all, if all you're going to do here is return from it?

Reply:
We just want to focus and zoom around a Node that is being selected by the User through the UI of the extension. ShowElement() ultimately triggers CenterViewOnElement() method from the WorkspaceView.xaml.cs

![image](https://user-images.githubusercontent.com/5354594/173924089-80280fa8-f2bc-456b-a3e0-cb8dde6228ed.png)